indexing-service/src/main/java/org/apache/druid/indexing/overlord/autoscaling/PendingTaskBasedWorkerProvisioningConfig.java

```diff
@@ -29,6 +29,8 @@
   @JsonProperty
   private int maxScalingStep = 10;
 
+  @JsonProperty
+  private Integer workerCapacityFallback = null;
```

Comment:
why did you choose to use an `Integer` object instead of an `int` with a default value?

I suspect a Nullable Integer may be more error prone as callers may not think to check for null and handle that case. At the very least, this field and `getWorkerCapacityFallback` should be annotated with `Nullable` so static analysis can catch places where users forget to add null checks.

Reply:
I wanted to use null as value not set. I guess I can also use -1 to do the same thing (value not set and do current behavior)

Reply:
Changed to int and use `-1` as default value to avoid potential NPE